pkg/ip/link.go

```diff
@@ -66,27 +73,76 @@ func makeVeth(name string, mtu int) (peerName string, veth netlink.Link, err err
 	return
 }
 
-// RandomVethName returns string "veth" with random prefix (hashed from entropy)
-func RandomVethName() (string, error) {
-	entropy := make([]byte, 4)
+const ENTROPY_LEN = 4
+
+// RandomIfaceName returns string @prefix with random suffix (hashed from entropy)
+func RandomIfaceName(prefix string) (string, error) {
+	if len(prefix)+ENTROPY_LEN > syscall.IFNAMSIZ {
```

Comment:
I'd like to discuss a counter-based naming scheme. Persisting the counter per-prefix should make this deterministic. We could calculate the maximum number of interfaces with `syscall.IFNAMSIZ - len(prefix)`.